src/lib/raft/raft.h

```diff
@@ -315,6 +315,10 @@ raft_process_heartbeat(struct raft *raft, uint32_t source);
 void
 raft_cfg_is_enabled(struct raft *raft, bool is_enabled);
 
+/** Make the instance a candidate. */
+void
+raft_start_candidate(struct raft *raft);
```

Comment:
This isn't right. This is an internal Raft method. Raft behaviour must be manipulated either via the config `raft_cfg_...` methods or via the Rafts high level methods affecting its state temporarily. For example,  "restore", "resign", "promote" - they are high-level methods making sense for the Raft algorithm. The "start candidate" looks odd, because 1) there is no "stop candidate" (there is, but it is still internal), 2) you are calling in your code essentially stop + start when you do `raft_resign` + `raft_start_candidate`.

The combination of this pair of calls I can't grasp right away. What exactly are you trying to achieve by this? Does the comment on `box_raft_leader_step_off` mean, that you will start new term after the death timeout passes since the latest WAL error?

If these couple of calls do what I understood, then it also seems you can do it as `raft_resign` + `raft_restore`, can you? The latter call will sync your cfg with the actual ability to be a candidate. Which will basically make it a candidate again in this case. And then you don't need to export any new methods from raft.

Reply:
Ok, I understand the concept, it's convenient. Thanks! Done.